MapboxNavigation/NavigationViewController.swift

```diff
@@ -201,15 +201,22 @@ open class NavigationViewController: UIViewController, NavigationStatusPresenter
      - parameter route: The route to navigate along.
      - parameter options: The navigation options to use for the navigation session.
      */
-    required public init(for route: Route,
+    required public init(for response: RouteResponse,
                          options: NavigationOptions? = nil) {
         super.init(nibName: nil, bundle: nil)
         
+        let route = response.routes!.first! //TODO: Safe-ish, but is there a better way?
```

Comment:
Such logic prevents picking alternative routes for navigation.
`RouteResponse` is useful as calculation result, but for configuring `NavigationViewController` I think it might be  not correct. Such design suggests that user has to pass in the result of a calculation, optionally modifying it's state, which comes beyond original `RouteResponse role. 
For this initializer, I don't think we can get far away from
`init(for route: Route, routeOptions: DirectionsOptions, options: NavigationOptions? = nil`
unless we introduce some new "OptionsWrapper" struct to contain everything needed.

Otherwise we force user to construct his own `RouteResponse` with adjusted properties.

Reply:
It's a known assumption that the first route in the `routes` model is the `activeRoute`. We allow for mutation of the `RouteResponse` object that allows the developer to customize which route they want to pick.